decksite/achievements.py

```diff
@@ -0,0 +1,237 @@
+from typing import TYPE_CHECKING, Dict, List, Optional
+
+from flask import url_for
+from flask_babel import ngettext
+
+import decksite
+from decksite.data import query
+from magic import tournaments
+
+if TYPE_CHECKING:
+    from decksite.data import person # pylint:disable=unused-import
+# Disabling unused-import supposedly not needed here but actually seems to be?
+
+def load_query(people_by_id: Dict[int, 'person.Person'], season_id: Optional[int]) -> str:
+    columns = ', '.join(f'SUM({a.key}) as {a.key}' for a in Achievement.all_achs if a.in_db)
+    return """
+        SELECT
+            person_id AS id,
+            {columns}
+        FROM
+            _achievements AS a
+        WHERE
+            person_id IN ({ids}) AND ({season_query})
+        GROUP BY
+            person_id
+    """.format(columns=columns, ids=', '.join(str(k) for k in people_by_id.keys()), season_query=query.season_query(season_id))
+
+def preaggregate_query() -> str:
+    create_columns = ', '.join(f'{a.key} INT NOT NULL' for a in Achievement.all_achs if a.in_db)
```

Comment:
We should probably backtick-escape (in case we use a SQL reserved word) and/or sql-escape (for safety) these when we interpolate them into the string unless we're like super duper sure that's unnecessary and no one will ever fuck that up in the future.